torch/_decomp/decompositions.py

```diff
@@ -2578,7 +2578,12 @@ def get_x_interp(y):
         return _upsample_cubic_interp1d(coeffs_x, t_x)
 
     coeffs_y = tuple((get_x_interp(y_ofs) for y_ofs in iys_ofs))
-    return _upsample_cubic_interp1d(coeffs_y, t_y)
+    result = _upsample_cubic_interp1d(coeffs_y, t_y)
```

Comment:
cc @fdrocha

Reply:
Thanks for the ping. This LGTM logic wise, but could be hard for compilers to optimize. It would be better to generate the result in the right memory format to begin with.
Shouldn't be too hard, but not obvious either. LMK if you will handle it, otherwise I can have a go.

Reply:
I can take a look that but not in this PR.

Reply:
Great. Separate PR makes sense

Reply:
According to conversation with @lezcano and @fdrocha, finally there is no particular point to set right memory format since the begining: `a[N_idx, C_idx, y_idx, x_idx]`. 

> Final copy (`result = result.contiguous(memory_format=memory_format)`) will be elided later on probably. Inductor takes intermediary strides more as a suggestion than something that must be hardcoded

Closing this discussion and no follow-up PR to be done.